tests/monotouch-test/Network/NWTxtRecordTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿#if !__WATCHOS__
+using System;
+using System.Collections.Generic;
+using System.Threading;
+#if XAMCORE_2_0
+using CoreFoundation;
+using Foundation;
+using Network;
+using ObjCRuntime;
+using Security;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.Network;
+using MonoTouch.Security;
+#endif
+
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.Network
+{
+	[TestFixture]
+	[Preserve(AllMembers = true)]
+	public class NWTxtRecordTest
+	{
+		NWTxtRecord record;
+		string randomKey = "MyData";
+		[SetUp]
+		public void SetUp ()
+		{
+			record = NWTxtRecord.CreateDictionary ();
+            record.SetValue (randomKey, new byte[3] { 0, 0, 0 });
+		}
+
+		[Test]
+		public void TestFromBytes ()
+		{
+			// get the raw data from the dictionary create txt record, and recreate a new one
+			ReadOnlyMemory<byte> rawData = null;
+			record.GetRawBytes (
+				(d) => {
+					rawData = d;
+				}
+			);
+            Assert.AreNotEqual (0, rawData.Length, "Raw data length.");
+			// do the required conversion
+			using (var otherRecord = NWTxtRecord.FromBytes (rawData))
+	            Assert.IsTrue (record.Equals (otherRecord), "Equals");
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			record?.Dispose ();
```

Comment:
`record` can't be `null` or `SetUp` would throw an NRE

Reply:
I can't but I don't trust people that change the code to be honest.

Reply:
I'd rather have the test fail if someone made such a change without running them - clearly worth investigating